internal/ingress/controller/controller.go

```diff
@@ -164,12 +166,14 @@ func (n *NGINXController) syncIngress(item interface{}) error {
 		PassthroughBackends: passUpstreams,
 	}
 
-	if !n.isForceReload() && n.runningConfig.Equal(&pcfg) {
-		glog.V(3).Infof("skipping backend reload (no changes detected)")
-		return nil
+	if !n.cfg.DynamicReload {
```

Comment:
The configuration could change, like an update to the configmap. This should be evaluated and reload NGINX even with the dynamic reload feature enabled.
To test this you can add/change `enable-vts: "true"` in the configuration configmap. You will see the diff of the nginx.conf (using the flag `--v=2`) but there's no NGINX reload.